biosys/apps/main/api/views.py

```diff
@@ -272,8 +273,64 @@ def get_serializer_context(self):
                 ctx['species_mapping'] = self.species_facade_class().name_id_by_species_name()
         return ctx
 
+    def search_data(self, qs, search_param, order_by=None):
+        """
+        Search does not support searching within JSONField.
+        :param qs: queryset
+        :param search_param: value to search
+        :param order_by: field to order by, prefixed with '-' for descending order
+        :return: the queryset after search filters applied
+        """
+        field_names = [field['name'] for field in self.dataset.data_package['resources'][0]['schema']['fields']]
+
+        where_clauses = []
+        params = []
+        for field_name in field_names:
+            where_clauses.append('data->>%s ILIKE %s')
+
+            params += [field_name, '%' + search_param + '%']
+
+            if order_by is not None and (order_by == field_name or order_by == '-' + field_name):
+                if order_by.startswith('-'):
+                    qs = qs.order_by(RawSQL('data->>%s', (order_by[1:],)).desc())
+                else:
+                    qs = qs.order_by(RawSQL('data->>%s', (order_by,)))
+
+        return qs.extra(where=['OR '.join(where_clauses)], params=params)
+
+    def order_by_data_field(self, qs, ordering_param):
+        """
+        Order by does not support ordering within JSONField.
+        :param qs: queryset
+        :param ordering_param: field to order by, prefixed with '-' for descending order
+        :return: the queryset after ordering is applied if order_by param is within data
+        """
+        field_names = [field['name'] for field in self.dataset.data_package['resources'][0]['schema']['fields']]
+
+        for field_name in field_names:
+            if ordering_param == field_name or ordering_param == '-' + field_name:
+                if ordering_param.startswith('-'):
+                    qs = qs.order_by(RawSQL('data->>%s', (ordering_param[1:],)).desc())
+                else:
+                    qs = qs.order_by(RawSQL('data->>%s', (ordering_param,)))
+
+        return qs
```

Comment:
maybe extract a util method for the ordering? To be used by Hydra.
Do we need to iterate the fields? What happen if you add an order for a key that doesn't exist? It raises an error?

Reply:
Yeah, I was thinking about that. If we're to have this same functionality with other views then I'll make it a utils function. I think we need to iterate the fields in the case of a search. If you add an order for a key that doesn't exist it will just be ignored, as it should be in my opinion.